crates/core/src/db/relational_db.rs

```diff
@@ -327,9 +327,12 @@ impl RelationalDB {
                     truncates: Box::new([]),
                 }),
             };
-            log::trace!("append {txdata:?}");
-            // TODO: Should measure queuing time + actual write
-            durability.append_tx(txdata);
+
```

Comment:
```suggestion
// Fix a minor bug where completely empty transactions would still be
// written to the commitlog. The bug is minor because, once we start
// logging inputs, all transactions will be non-empty.

// The check is done in relational DB rather than the durability crate,
// because in principle empty transactions are permissible, and may be used
// in the future (e.g. to confirm a certain offset).
```